lib/notifier.js

```diff
@@ -94,20 +94,39 @@ function buildBaseData(extra) {
 
 
 function buildErrorData(baseData, err, callback) {
-  parser.parseException(err, function (e, errData) {
-    if (e) {
-      return callback(e);
-    }
-    baseData.body.trace = {
-      frames: errData.frames,
-      exception: {
-        class: errData['class'],
-        message: errData.message
-      }
-    };
-
-    callback(null);
-  });
+  var errors = [];
+    
+    do {
+        errors.push(err);
+    } while((err = err.innerError) !== undefined)
+
+    baseData.body.trace_chain = [];
+
+    var chain = baseData.body.trace_chain;
+    async.eachSeries(errors, function(ex, cb) {
+        
+        parser.parseException(ex, function (e, errData) {
+            if (e) {
+                if(chain.push(e) == 1) cb(e);
+            }
+                
+            chain.push({
+                frames: errData.frames,
+                exception: {
+                    class: errData['class'],
+                    message: errData.message
+                }
+            });
+
+            cb();
+        });
+
+    }, function(err) {  
+        if(err) 
```

Comment:
You will want to use `return callback(err);` here so we don't call the callback twice.

Style: always enclose conditional bodies in `{}`.